ibis/backends/clickhouse/compiler/values.py

```diff
@@ -1355,16 +1355,49 @@ def _last_value(op, **kw):
 }
 
 
-@translate_val.register(ops.ParseURL)
-def _parse_url(op, **kw):
-    if (func := _PARSE_URL_FUNCS.get(extract := op.extract)) is None:
-        raise ValueError(
-            f"`{extract}` is not supported in the parse_url implementation for the "
-            "ClickHouse backend"
+@translate_val.register(ops.ExtractProtocol)
+def _extract_protocol(op, **kw):
+    arg = translate_val(op.arg, **kw)
+    return f"CAST(nullIf(protocol({arg}), '') AS Nullable(String))"
```

Comment:
I think all the `CAST`s can be removed as `nullif` appears to have a `Nullable(String)` return type:

```
play-us :) explain header = 1 select nullif('a', '');

EXPLAIN header = 1
SELECT nullif('a', '')

Query id: d6de3f0f-2616-43ce-af61-6a1f1728a5a9

┌─explain─────────────────────────────────────┐
│ Expression ((Projection + Before ORDER BY)) │
│ Header: nullIf('a', '') Nullable(String)    │
│   ReadFromStorage (SystemOne)               │
│   Header: dummy UInt8                       │
└─────────────────────────────────────────────┘

4 rows in set. Elapsed: 0.001 sec.
```

Reply:
Done